crates/p2p/src/tests/setup_invalid_signature.rs

```diff
@@ -0,0 +1,203 @@
+//! Tests for the setup phase of P2P connections where signature is invalid.
+
+use std::time::Duration;
+
+use libp2p::{
+    PeerId, build_multiaddr,
+    identity::{Keypair, PublicKey},
+};
+use tokio::{sync::oneshot::channel, time::sleep};
+use tokio_util::sync::CancellationToken;
+use tracing::info;
+
+use crate::{
+    commands::{Command, ConnectToPeerCommand, QueryP2PStateCommand},
+    signer::ApplicationSigner,
+    tests::common::{MockApplicationSigner, User, init_tracing},
+};
+
+#[derive(Debug, Clone)]
+struct BadApplicationSigner;
+
+impl BadApplicationSigner {
+    pub(crate) fn new(_app_keypair: Keypair) -> Self {
+        Self
+    }
+}
+
+impl ApplicationSigner for BadApplicationSigner {
+    fn sign(
+        &self,
+        _message: &[u8],
+        _app_public_key: PublicKey,
+    ) -> Result<Vec<u8>, Box<dyn std::error::Error + Send + Sync>> {
+        let signature = Vec::from([0x02; 32]);
+        Ok(signature)
```

Comment:
This is why we don't do `Vec<u8>` for stuff we know the size.
The signature is 64 bytes for `ed25519`.
How do I know that this test is not failing for wrong sig size or really invalid signature?

Can you change this to `[2u8; 64]` and removing the `Vec<u8>`?